pkg/usersettings/handlers.go

```diff
@@ -0,0 +1,340 @@
+	path := fmt.Sprintf("/apis/user.openshift.io/v1/users/~")
+	url := proxy.SingleJoiningSlash(h.Endpoint, path)
+
+	request, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", user.Token))
+	request.Header.Set("Accept", "application/json")
+
+	response, err := h.Client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("getCurrentUser %d %s\n", response.StatusCode, response.Status)
+
+	if response.StatusCode != 200 {
+		return nil, errors.New(fmt.Sprintf("Unexpected status code (%d instead of 200) when getting current user", response.StatusCode))
+	}
+
+	currentUser := User{}
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("getCurrentUser body: %s\n", string(body))
+
+	err = json.Unmarshal(body, &currentUser)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("getCurrentUser currentUser: %v\n", currentUser)
+
+	return &currentUser, nil
+}
+
+func (h *UserSettingsHandler) createRole(role *Role) (*http.Response, error) {
+	path := fmt.Sprintf("/apis/rbac.authorization.k8s.io/v1/namespaces/%s/roles", role.Metadata.Namespace)
+	return h.sendRequest(http.MethodPost, path, role)
+}
+
+func (h *UserSettingsHandler) createRoleBinding(roleBinding *RoleBinding) (*http.Response, error) {
+	path := fmt.Sprintf("/apis/rbac.authorization.k8s.io/v1/namespaces/%s/rolebindings", roleBinding.Metadata.Namespace)
+	return h.sendRequest(http.MethodPost, path, roleBinding)
+}
+
+func (h *UserSettingsHandler) createConfigMap(configMap *ConfigMap) (*http.Response, error) {
+	path := fmt.Sprintf("/api/v1/namespaces/%s/configmaps", configMap.Metadata.Namespace)
+	return h.sendRequest(http.MethodPost, path, configMap)
+}
+
+func (h *UserSettingsHandler) getConfigMap(namespace string, name string) (*http.Response, error) {
+	path := fmt.Sprintf("/api/v1/namespaces/%s/configmaps/%s", namespace, name)
+	return h.sendRequest(http.MethodGet, path, nil)
+}
+
+func (h *UserSettingsHandler) deleteConfigMap(namespace string, name string) (*http.Response, error) {
+	path := fmt.Sprintf("/api/v1/namespaces/%s/configmaps/%s", namespace, name)
+	return h.sendRequest(http.MethodDelete, path, nil)
+}
+
+func (h *UserSettingsHandler) deleteRole(namespace string, name string) (*http.Response, error) {
+	path := fmt.Sprintf("/apis/rbac.authorization.k8s.io/v1/namespaces/%s/roles/%s", namespace, name)
+	return h.sendRequest(http.MethodDelete, path, nil)
+}
+
+func (h *UserSettingsHandler) deleteRoleBinding(namespace string, name string) (*http.Response, error) {
+	path := fmt.Sprintf("/apis/rbac.authorization.k8s.io/v1/namespaces/%s/rolebindings/%s", namespace, name)
+	return h.sendRequest(http.MethodDelete, path, nil)
+}
+
+func (h *UserSettingsHandler) sendRequest(method string, path string, content interface{}) (*http.Response, error) {
```

Comment:
Is there a reason we can't use client-go instead of rolling our own? We should have the library already as a dependency.

Reply:
I replaced all http request calls with the `client-go` library, incl. fetching the ConfigMap. So we can not "proxy" the response here, but can just return the 'parsed' ConfigMap as json again.